src/vm/fieldmarshaler.cpp

```diff
@@ -3910,7 +3910,27 @@ VOID FieldMarshaler_Delegate::UpdateNativeImpl(OBJECTREF* pCLRValue, LPVOID pNat
     }
     CONTRACTL_END;
 
+    // A cleanup list MUST be specified in order for us to be able to marshal
+    // the Delegate.
```

Comment:
Is this also used for manual marshaling using `PtrToStructure` - is this change going to break it?

Reply:
I will check and verify for you.

Reply:
Yes it does.

I have two possible ideas for this:

- Add a CleanupWorkList in the implementation of `Marshal.StructureToPtr`. I don't think this one really makes sense but it's an option.
- Make having a CleanupWorkList optional for delegate marshalling. This would make more sense IMO.

Reply:
> Make having a CleanupWorkList optional for delegate marshalling. This would make more sense IMO.

Agree. (Also, check that we have a test for this - this test should be failing with the current version of the change.)

Reply:
Added a test om the latest commit.